src/index.ts

```diff
@@ -335,41 +442,31 @@ export function offlineInit(config: IClientConfigSync): EppoClient {
     }
   }
 
-  EppoJSClient.initialized = true;
-  return EppoJSClient.instance;
+  instance.initialized = true;
+  return instance;
 }
 
-/**
- * Tracks pending initialization. After an initialization completes, the value is removed from the map.
- */
-let initializationPromise: Promise<EppoClient> | null = null;
-
 /**
  * Initializes the Eppo client with configuration parameters.
  * This method should be called once on application startup.
  * If an initialization is in process, calling `init` will return the in-progress
  * `Promise<EppoClient>`. Once the initialization completes, calling `init` again will kick off the
  * initialization routine (if `forceReinitialization` is `true`).
```

Comment:
unrelated to the PR but I just noticed that the field is called `forceReinitialize`
```suggestion
 * initialization routine (if `forceReinitialize` is `true`).
```